Makefile

```diff
@@ -187,41 +185,30 @@ build-all: $(addprefix sub-build-,$(VALIDARCHES))
 sub-build-%:
 	$(MAKE) build ARCH=$*
 
-# Update the vendored dependencies with the latest upstream versions matching
-# our glide.yaml.  If there area any changes, this updates glide.lock
-# as a side effect.  Unless you're adding/updating a dependency, you probably
-# want to use the vendor target to install the versions from glide.lock.
-.PHONY: update-vendor
-update-vendor:
-	mkdir -p $$HOME/.glide
-	$(DOCKER_RUN) $(CALICO_BUILD) glide up --strip-vendor
-	touch vendor/.up-to-date
-
 # vendor is a shortcut for force rebuilding the go vendor directory.
 .PHONY: vendor
 vendor: vendor/.up-to-date
-vendor/.up-to-date: glide.lock
-	mkdir -p $$HOME/.glide
-	$(DOCKER_RUN) $(CALICO_BUILD) glide install --strip-vendor
+vendor/.up-to-date: go.mod
```

Comment:
Does Typha still need the vendor directory?

Reply:
Most (or all) of our repos rely on the vendor directory for local builds. As far as I can see that's the only reason we still need vendoring.

Reply:
Noncontainer builds should use go modules as well, in which case vendor isn't needed.  For example, after a `make clean` I can still run `go build -o calico-typha cmd/calico-typha/typha.go` and it builds fine.

Reply:
I meant LOCAL_BUILDS, or the ones that use ../libcalico-go (or other local repositories).

Reply:
Use a `replace` command in `go.mod` to point to a local source library, rather than the `vendor` directory.

Reply:
Ok, I think I got this working now. Also removed all the vendoring code. Please take a look (search for local_build in the Makefile).